HHS/Head-Start-TTADP
# [TTAHUB-956 ] Fix Clearing of Objective Title, Topics, and Roles

## Description of change

Fix Clearing of Objective Title, Topics, and Roles on AR.


## How to test

Case 1:
1. Create a new AR
2. Create  a new Goal
3. Add two Objectives and complete all info
4. Remove the first Objective
5. Make sure none of the previously entered information has been removed from the second objective

Case 2 (with saving):
1. Create a new AR
2. Create a new Goal
3. Add a single Objective complete all info and save
4. Save the Goal
5. Edit the goal via the context menu ...
6. Add another Objective and fill out all the info
7. Remove the first Objective originally saved
8. None of the info from the newly created Objective should have been removed

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-956


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
